EIPS/eip-6963.md

```diff
@@ -0,0 +1,298 @@
+        "id": "rfc3986",
+        "citation-label": "rfc3986",
+        "issued": {
+            "date-parts": [
+                [2005, 1]
+            ]
+        },
+        "number-of-pages": "61",
+        "publisher": "RFC Editor",
+        "title": "Uniform Resource Identifier (URI): Generic Syntax",
+        "URL": "https://www.rfc-editor.org/info/rfc3986"
+    }
+    ```
+
+### Provider Detail
+
+The wallet provider will also announce their own [EIP-1193](./eip-1193.md) provider interface in parallel with the provider info provided above in the following interface `EIP6963ProviderDetail`
+
+```typescript
+interface EIP6963ProviderDetail {
+  info: EIP6963ProviderInfo;
+  provider: EIP1193Provider;
+}
+```
+
+The `EIP1193Provider` interface is documented at [EIP-1193](./eip-1193.md) and can be used to override the `window.ethereum` object once the user has explicitly selected it.
+
+### Window Event
+
+Different wallet providers might inject their scripts at different times, plus there is no guarantees that the Ethereum library in the web page will be loaded before all injected scripts are present in the page.
+
+To emit events it must use `window.dispatchEvent` and to observe events it must use `window.addEventListener`. There will be two event interfaces:
+
+```typescript
+// Requesting an EVM provider
+interface EIP6963RequestProviderEvent extends Event {
+  type: "eip6963:requestProvider";
+}
+
+// Annoucing an EVM provider
+interface EIP6963AnnounceProviderEvent extends CustomEvent {
+  type: "eip6963:announceProvider";
+  detail: EIP6963ProviderDetail;
+}
+```
+
+Therefore both wallet providers and the Ethereum library must emit events when they initialize.
+
+```typescript
+// Ethereum library initializes
+window.dispatchEvent(new Event("eip6963:requestProvider"));
+
+// Wallet provider initializes
+window.dispatchEvent(new CustomEvent("eip6963:announceProvider", { detail }));
+```
+
+Additionally the Wallet providers will react to the event emitted by the Ethereum library.
+
+```typescript
+window.addEventListener(
+  "eip6963:requestProvider",
+  (event: EIP6963RequestProviderEvent) => {
+    announceProvider();
+  }
+);
+```
+
+## Rationale
+
+### Interfaces
+
+Standardizing a provider info interface (`EIP6963ProviderInfo`) allows determining the necessary information to populate a wallet selection popup. This is particularly useful for web3 onboarding libraries such as Web3Modal, RainbowKit, Web3Onboard, ConnectKit, etc.
+
+Regarding the announced provider interface (`EIP6963ProviderDetail`) it was important to leave the [EIP-1193](./eip-1193.md) provider interface untouched for backwards-compatiblity therefore it's exposed in parallel.
+
+### Identifiers
+
+A locally unique identifier prevents from conflicts using the same globally unique identifier by using UUID v4.0
+
+A globally unique identifier is used to for machine-readable detection of which wallet is being used and it can take different formats, for example:
```

Comment:
```suggestion
A globally unique identifier is used for machine-readable detection of the currently active wallet and it can take different formats, for example:
```